golang/x/relay/client/cli/tx.go

```diff
@@ -23,31 +25,62 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
 	}
 
 	relayTxCmd.AddCommand(client.PostCommands(
-		GetCmdSetLink(cdc),
+		GetCmdIngestHeaderChain(cdc),
 	)...)
 
 	return relayTxCmd
 }
 
-// GetCmdSetLink is the CLI command for sending a SetLink transaction
-func GetCmdSetLink(cdc *codec.Codec) *cobra.Command {
+// GetCmdIngestHeaderChain creates a CLI command to ingest a header chain
+func GetCmdIngestHeaderChain(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
-		Use:   "setlink [hex header] [signer address]",
+		Use:   "ingestheaders <json list of headers>",
 		Short: "Set a link",
-		Args:  cobra.ExactArgs(2),
+		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
 			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
 
-			addr := sdk.AccAddress(args[1])
-			msg := types.NewMsgSetLink(addr, args[0], cliCtx.GetFromAddress())
+			// note: unmarshalling json here is undesirable, but necessary
+			// 	     for MarkNewHeaviest we want to accept text instead of json
```

Comment:
`for IngestHeaderChain`

Reply:
no, this is an informational note about the future :)